spec/build-hooks-spec.js

```diff
@@ -0,0 +1,110 @@
+'use babel';
+
+import fs from 'fs-extra';
+import temp from 'temp';
+import specHelpers from 'atom-build-spec-helpers';
+
+describe('Hooks', () => {
+  let directory = null;
+  let workspaceElement = null;
+  let target = null;
+  const commandName = 'build:hook-test:trigger-build';
+
+  function getBuildModule() {
+    return atom.packages.getLoadedPackage('build').mainModule;
+  }
+
+  temp.track();
+
+  beforeEach(() => {
+    directory = fs.realpathSync(temp.mkdirSync({ prefix: 'atom-build-spec-' }));
+    atom.project.setPaths([ directory ]);
+
+    atom.config.set('build.buildOnSave', false);
+    atom.config.set('build.panelVisibility', 'Toggle');
+    atom.config.set('build.saveOnBuild', false);
+    atom.config.set('build.notificationOnRefresh', true);
+
+    jasmine.unspy(window, 'setTimeout');
+    jasmine.unspy(window, 'clearTimeout');
+
+    runs(() => {
+      workspaceElement = atom.views.getView(atom.workspace);
+      jasmine.attachToDOM(workspaceElement);
+    });
+
+    waitsForPromise(() => {
+      return atom.packages.activatePackage('build');
+    });
+
+    runs(() => {
+      fs.writeFileSync(directory + '/.atom-build.json', fs.readFileSync(__dirname + '/fixture/.atom-build.hooks.json'));
+    });
+
+    waitsForPromise(() => specHelpers.refreshAwaitTargets());
+  });
+
+  afterEach(() => {
+    fs.removeSync(directory);
+  });
+
+  it('should call preBuild', () => {
+    runs(() => {
+      target = getBuildModule().targets[directory].find(t => t.atomCommandName === commandName);
```

Comment:
Fair point.

In fact, I did initially intend to include small package for this tests, but I thought that it would be better to add a `.json` config as in all other tests to keep the tests consistent.